### PR TITLE
Update DG

### PR DESCRIPTION
**Pull request by tayruxin**
- add planned enhancement 
- add manual testing 
- edit activity diagram

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -672,6 +672,32 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### **1. More Specific Success Message for Company**
 
+**Feature Flaw in Current Implementation**
+
+Currently, the success message for view, edit, remark, unremark, and add commands only displays the company name.
```
Comment (GlendaChong) [inline docs/DeveloperGuide.md:677]: `view`, `edit`, `remark`, `unremark`, and `add`

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -672,6 +672,32 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+However, the duplicate check implemented in LinkMeIn uses 3 criteria, company name, role, and deadline. As such, there
+can be more than one entry with the same company name. The user might want to know the role and deadline to
+differentiate between applications with the same company name. As such, with the current implementation of the success
+message, it can be confusing for the user as to which company specifically has been modified.
```
Comment (GlendaChong) [inline docs/DeveloperGuide.md:681]: As such, the current success message displayed can be confusing for the user as to which company specifically has been modified.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -672,6 +672,32 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+**Proposed Enhancement**
+
+Instead of displaying only the company name, the success message will display the company name, role and deadline. This
+is the implementation that is done for the existing `delete` command. As such, a method was already created in
```
Comment (GlendaChong) [inline docs/DeveloperGuide.md:686]: This will be the same implementation done for the existing `delete` command.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -672,6 +672,32 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+The following implementation will be adopted instead:
+`return new CommandResult(String.format(MESSAGE_SUCCESS, Messages.getCompanyInfo(toAdd)));`
+
+**Examples**
+
+- **view**: `Viewing Company: COMPANY_NAME (ROLE, DEADLINE)`
+- **edit**: `COMPANY_NAME (ROLE, DEADLINE) company edited.`
+- **remark**: `Added remark to Company: COMPANY_NAME (ROLE, DEADLINE)`
+- **unremark**: `Removed remark from Company: COMPANY_NAME (ROLE, DEADLINE)`
+- **add**: `New company added: COMPANY_NAME (ROLE, DEADLINE)`
```
Comment (GlendaChong) [inline docs/DeveloperGuide.md:699]: - **view**: `Viewing company: COMPANY_NAME (ROLE, DEADLINE)`
- **edit**: `COMPANY_NAME (ROLE, DEADLINE) company edited.`
- **remark**: `Added remark to company: COMPANY_NAME (ROLE, DEADLINE)`
- **unremark**: `Removed remark from company: COMPANY_NAME (ROLE, DEADLINE)`
- **add**: `New company added: COMPANY_NAME (ROLE, DEADLINE)`


change the "Company" to small letters for standardisation

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -692,6 +718,27 @@ If the user did not add in the recruiter's name, phone number and email address
...
 ### **4. Enhanced Flexibility in Phone Number Parameter Input**
 
+**Feature Flaw in Current Implementation**
+
+Currently, the phone number field only accepts integers as valid user inputs. However, users may encounter scenarios,
+such as applying for overeseas internship, where they want to include symbols like `()`, `+`,  `-` and `.` in the phone
+number field. The current restriction prevents users from indiciating country codes, potentially causing confusion about
+the origin of the phone number.
+
+**Proposed Enhancement:**
```
Comment (GlendaChong) [inline docs/DeveloperGuide.md:728]: remove :

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -692,6 +718,27 @@ If the user did not add in the recruiter's name, phone number and email address
 
 ### **4. Enhanced Flexibility in Phone Number Parameter Input**
 
+**Feature Flaw in Current Implementation**
+
+Currently, the phone number field only accepts integers as valid user inputs. However, users may encounter scenarios,
+such as applying for overeseas internship, where they want to include symbols like `()`, `+`,  `-` and `.` in the phone
```
Comment (GlendaChong) [inline docs/DeveloperGuide.md:724]: overseas internships

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -692,6 +718,27 @@ If the user did not add in the recruiter's name, phone number and email address
 
 ### **4. Enhanced Flexibility in Phone Number Parameter Input**
 
+**Feature Flaw in Current Implementation**
+
+Currently, the phone number field only accepts integers as valid user inputs. However, users may encounter scenarios,
```
Comment (GlendaChong) [inline docs/DeveloperGuide.md:723]: Change "field" to "parameter"

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -692,6 +718,27 @@ If the user did not add in the recruiter's name, phone number and email address
...
+such as applying for overeseas internship, where they want to include symbols like `()`, `+`,  `-` and `.` in the phone
+number field. The current restriction prevents users from indiciating country codes, potentially causing confusion about
+the origin of the phone number.
+
+**Proposed Enhancement:**
+
+The regex checking for a valid phone number will be changed to allow for `()`, `+`, `-` and `.` in the phone number
+field. In addition, the `+` will only be allowed at the start while, the other symbols have no positioning restrictions.
```
Comment (GlendaChong) [inline docs/DeveloperGuide.md:731]: the `+` character/symbol?

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -725,6 +772,22 @@ Also, we understand that some users may not wish to type leading zeros for days
 
 ### **9. Allow Multiple Indexes Input for Delete Command**
 
+**Feature Flaw in Current Implementation**
+
+Currently, users can only delete one entry at once. However, there will be cases where the user wish to delete multiple entries at once, especially if the user wishes to delete all the companies that he got rejected from. This can be tedious and inconvenient for the user.
```
Comment (GlendaChong) [inline docs/DeveloperGuide.md:777]: change entry to company

also standardise between users/the user?

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -725,6 +772,22 @@ Also, we understand that some users may not wish to type leading zeros for days
...
+
+**Proposed Enhancement**
+
+Enable users to input multiple indices when attempting to delete entries. Users can separate each index with a comma. The delete command parser will then split the string by commas and remove the companies corresponding to the specified indices. There will also be checks to see if the user keyed in the same index more than once. If the same index is keyed in more than once, the parser will accept the input but treat it as if the user only keyed in that same index once.
```
Comment (GlendaChong) [inline docs/DeveloperGuide.md:781]: standardise indices/indexes with header -- either one is acceptable

Comment (GlendaChong) [inline docs/DeveloperGuide.md:781]: `DeleteCommandParser`

File: docs/DeveloperGuide.md [hunk 11]
```diff
@@ -725,6 +772,22 @@ Also, we understand that some users may not wish to type leading zeros for days
...
+
+**Examples**
+
+- `delete 1, 2`: deletes company at index 1 and 2
+- `delete 1`: deletes company at index 1
+- `delete 4, 3, 7, 2`: deletes company at index 4, 3, 7, 2
```
Comment (GlendaChong) [inline docs/DeveloperGuide.md:787]: companies for those with more than 1?

File: docs/DeveloperGuide.md [hunk 12]
```diff
@@ -762,22 +825,46 @@ testers are expected to do more *exploratory* testing.
...
+
+1. Viewing a company while all the companies are being shown
+    1. Prerequisites: List all companies using the `list` command. Multiple companies in the list.
+    2. Test case: `view 1`<br>
+       Expected: First company is shown in the company detail panel.
+    3. Test case: `view 0`<br>
+       Expected: No company is shown in the company detail panel. Error details shown in the message box.
+    4. Other incorrect view commands to try: `view`, `view x`, `...` (where x is larger than the list size)<br>
+       Expected: Similar to previous.
+
```
Comment (GlendaChong) [inline docs/DeveloperGuide.md]: not sure how we want to standardise with my pr as well. 

are we doing it very detailed?

File: docs/images/CompanyDetailPanelActivityDiagram.png
Comment (GlendaChong) [inline docs/images/CompanyDetailPanelActivityDiagram.png:1]: checkDelete(company), could that part be changed to words?

**Review by GlendaChong — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md (hunk 9 above)
Comment (tayruxin) [reply docs/DeveloperGuide.md:777]: we are using the user?

**Review by GlendaChong — APPROVED** [review]
lgtm

**Pull request merged**